### DISCUSSION TITLE
How to list all documents with attachment

### DISCUSSION THREAD

**Discussion by hshenMcc** [ravendb/ravendb#15829] [Q&A] [answered]
How may I make a RQL query to only get documents with attachment? 
I tried
   from "@all_docs" w where w.@metadata.@flags = "HasAttachments"
It returns nothing

**Comment by ayende** [comment, marked as answer]
`@flags` isn't indexable, I'm afraid. use:

`from "@all_docs" w where exists(w.@metadata.@attachments)`

**Reply by hshenMcc** [reply]
Works. Thanks.
